### PR TITLE
(#518) Align nci primary nav to usa primary primary nav 

### LINKED ISSUE #492: At 1024 the menu goes wonky and alignment is off
### Issue description
At the 1024 breakpoint, the menu goes wonky, and alignment is off.

> **ESTIMATE** 3

### Steps to reproduce the issue

1.  Go to storybook https://designsystem-dev.cancer.gov/pr-455/storybook/iframe.html?id=components-nci-header--nci-extended-desktop-longer-primary-item&globals=backgrounds.grid:false&viewMode=story
2. Open nci-extended-desktop-longer-primary-item
3. If needed, open in new window to see at 1024 breakpoint

### What's the expected result?
COMP: https://publicis.invisionapp.com/overview/NCI-DS-Spanish-cl2qe0hwx001r01aa3ajtac6t/screens
- Menu items don't wrap at 1024 (per comp with actual content, nav labels don't wrap) [Bug]
- Menu items should all be center aligned horizontally [Change Request]

### What's the actual result?
- Menu items wrap at 1024
- Menu items are top aligned

### Additional details / screenshot
-Screen Shot 2022-06-15 at 3 53 59 PM

### Related Tickets
* Issue #455

### LINKED ISSUE #512: NCIDS Header Does not allow for setting a menu link's current state
Not every item will have a mega menu and thus we need to be able to set the "on" state (`usa-current`) for a primary nav. This is not possible now as it only works for mega menu buttons.

See the screen shot below when on the About NCI page
image

This was for the markup:
(Note: I even tried adding `usa-current` to the `li` tag.)
image

### LINKED ISSUE #518: Fix the Primary Navigation on the NCI Header to be more like USWDS
>**TL;DR;** We did not follow USWDS see [What Needs to be Done](https://github.com/NCIOCPL/ncids/issues/518#issuecomment-1195584541).

This ticket is now a rollup of all the Primary Nav issues that have the same root cause. We strayed too far from the USWDS for the primary nav of the NCI Header.

### Limited Nav Items Look Weird

Bryan found an issue where NCI CONNECT Spanish site only has 2 top nav items and the two nav items are seen on the opposite page of each other (see screenshots)

DS:
image (18)

PROD:
image (19)

Need to look at different sites on the platform that have smaller site nav:
| Site  | # of Nav Items |
| ------------- | ------------- |
| Cancer.gov ([English](https://www.cancer.gov/) & [Spanish](https://www.cancer.gov/espanol)) (English & Spanish)  | 6 |
| [NCI CONNECT](https://www.cancer.gov/rare-brain-spine-tumor/)  | 5  |
| [NCI CONNECT (Spanish)](https://www.cancer.gov/rare-brain-spine-tumor/espanol)  | 2  |
| [MyPART](https://www.cancer.gov/pediatric-adult-rare-tumor/)  | 6  |
| [DCEG](https://dceg.cancer.gov/)  | 5  |
| [Physics](https://physics.cancer.gov/)  | 5  |
| [Nano](https://www.cancer.gov/nano/)  | 5  |
| [CCG](https://ccg-cms.cancer.gov/ccg/)  | 4  |
| [SBIR](https://sbir-cms.cancer.gov/)  | 5  |

> **ESTIMATE** TBD

### Steps to reproduce the issue

1.  Go to https://www.cancer.gov/rare-brain-spine-tumor/espanol and compare the nav spacing in the mega menu to DS screenshot above

### What's the expected result?
- To align with USWDS nav items should align left and add padding based on USWDS: https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-header--extended-megamenu&args=&viewMode=story (per Bryan it's 1rem of padding

Screen shot of cancer.gov nav items with this expected result:
image (23)

### What's the actual result?
- the nav items are not left aligned

### Additional details / screenshot
- see above

### Related Tickets
* #436 
* #492 
* [#3277](https://github.com/NCIOCPL/cgov-digital-platform/issues/3277)
* [#3268](https://github.com/NCIOCPL/cgov-digital-platform/issues/3268)

### PR DESCRIPTION
**Pull request by olitharp-nci**
Closes #518
Closes #512 
Closes #492 maybe

## Viewing Information
* [The documentation site](https://designsystem-dev.cancer.gov/pr-545/)
* [The storybook site](https://designsystem-dev.cancer.gov/pr-545/storybook/)
* [The example site](https://designsystem-dev.cancer.gov/pr-545/example-site/)

### CODE REVIEW

**Review by karlikpj — APPROVED** [review]

**Comment by bennettcc** [comment]
@olitharp-nci @bryanpizzillo  Per 8/23 sprint demo client feedback - please top align the megamenu items to make them consistent with the DS standards. Currently, they are center aligned (to match the original comps)

cc @meglake

**Review by laurelthrash — APPROVED** [review]
PR approved on behalf of product. Of note, demo'd the header with the federal team during meeting on 08/25 where the header was approved.

**Review by bryanpizzillo — APPROVED** [review]
